src/viam/sdk/config/resource.cpp

```diff
@@ -18,11 +19,7 @@ namespace viam {
 namespace sdk {
 
 Name ResourceConfig::resource_name() {
-    try {
-        this->fix_api();
-    } catch (std::string err) {  // NOLINT
-        throw err;
-    }
```

Comment:
Does this do something that I don't understand?

Reply:
Probably not! Looks like the `try/catch` is pretty much meaningless there, and catching a `str` is just an artifact of poor initial design decisions. Makes sense to remove.